_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+          .then(() => {
+            this.message = "";
+          })
+          .catch((err) => {
+            return console.error(err.toString());
+          });
+      },
+    },
+  };
+
+  Vue.createApp(VueChatApp).mount("#chatApp");
+</script>
+```
+
+{% endraw %}
+
+We use CDN for the SignalR client library, and the connection to the ChatHub uses the endpoint you set earlier:
+
+```js
+const connection = new signalR.HubConnectionBuilder().withUrl("/chatHub").build();
+```
+
+The state is stored in the data section of our Vue app. There are three properties in use; A flag to let us know if it is connected, the message currently being typed, and a chat log.
+
+```js
+data() {
+  return {
+    isConnected: false,
+    message: "",
+    chatLog: [],
+  }
+},
+```
+
+Your app has two methods: `receiveMessage` and `sendMessage`.
+
+When `receiveMessage` is called, it just appends an object to the chat log with the user and the message.
+
+When `sendMessage` is called, we use the SignalR connection to invoke "SendMessage" and pass along our message properties. Once the message is sent, we blank it out to fill a new message.
+
+```js
+receiveMessage(user, message) {
+  this.chatLog.push({
+    User: user,
+    Message: message,
+  });
+},
+sendMessage() {
+  connection
+    .invoke("SendMessage", this.message)
+    .then(() => {
+      this.message = "";
+    })
+    .catch((err) => {
+      return console.error(err.toString());
+    });
+},
+```
+
+When the app is created, a hook is added for "ReceiveMessage" that calls the ViewModel `receiveMessage` method described previously.
+
+Then the connection to the hub is started, and if successful, `isConnected` is set to true.
+
+```js
+connection.on("ReceiveMessage", (user, message) => {
+  this.receiveMessage(user, message);
+});
+connection
+  .start()
+  .then(() => {
+    this.isConnected = true;
+  })
+  .catch((err) => {
+    return console.error(err.toString());
+  });
+```
+
+If you run your application at this point, you'll have a secured chat application running in a container!
+
+## Deploy Your .NET Chat Application to AWS
+
+Now that the chat application is complete, it's time to deploy it. First, we need to build the docker image and deploy it to AWS to be used. AWS has private container repositories via its Elastic Container Registry (ECR) product.
+
+### Create an ECR Repository
+
+First, we need to make one final change to the **Dockerfile** to get TLS working on Fargate.
+
+Add the following two lines to the **Dockerfile** right after the `dotnet publish` command:
+
+```dockerfile
+# build and publish the app
+...
+RUN dotnet publish -c release -o /app --no-restore
+
+ARG CERT_PASSWORD
+RUN dotnet dev-certs https -ep /app/aspnetapp.pfx -p ${CERT_PASSWORD}
+
+# final stage/image
+...
+```
+
+This will create a self-signed development certificate right within the Docker image, which we can use to run the application with TLS on Fargate for demo purposes.
```

Comment:
```suggestion
This will create a self-signed development certificate right within the docker image, which we can use to run the application with TLS on Fargate for demo purposes.
```
aiming for consistent use of lowercase for docker image, docker file, docker container, as devs often do.
For naming the platform/product overall I'm keeping Docker uppercase.